azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.execapp;
+
+import azkaban.executor.ExecutableFlow;
+import azkaban.project.ProjectFileHandler;
+import azkaban.project.ProjectLoader;
+import azkaban.project.ProjectManagerException;
+import azkaban.utils.FileIOUtils;
+import azkaban.utils.Pair;
+import azkaban.utils.Utils;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.Map;
+import java.util.zip.ZipFile;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+
+import static com.google.common.base.Preconditions.*;
+import static java.util.Objects.*;
+
+
+public class FlowPreparer {
```

Comment:
I'm not sure what to add in the javadoc specifically.

If I add `FlowPreparer` prepares the flow or something: this is a perhaps redundant. I tried to name the methods and classes to be as self explanatory as possible so that documentation required is minimal. Also, I assume that these pieces of code will go through a lot of churn till they are stable. Documentation might be an overhead to maintain whilst this change. Should we add this later if at all? I don't see this to be an external facing API anyway.